communication/src/main/java/org/jboss/da/communication/pom/PomAnalyzer.java

```diff
@@ -0,0 +1,246 @@
+
+                        // convert from ProjectVersionRef to GAV
+                        GAV originGAV = generateGAV(origin);
+                        GAV targetGAV = generateGAV(target);
+
+                        // get the origin GAVDependencyTree
+                        GAVDependencyTree originScmGAVTree = addGAVDependencyTreeToMapper(
+                                gavDependencyTreeMap, originGAV);
+                        GAVDependencyTree targetScmGavTree = addGAVDependencyTreeToMapper(
+                                gavDependencyTreeMap, targetGAV);
+
+                        if (rootRelationship)
+                            root = originScmGAVTree;
+
+                        addTargetToGAVDependencyTree(originScmGAVTree, targetScmGavTree,
+                                relationship);
+                    }
+                }
+
+                return root;
+
+            } finally {
+                // cleanup
+                FileUtils.deleteDirectory(tempDir);
+            }
+        } catch (TransferException | GalleyMavenException | URISyntaxException | CartoDataException
+                | IOException e) {
+            throw new PomAnalysisException(e);
+        }
+    }
+
+    /**
+     * Return an existing GAVDependencyTree if present in the map, otherwise create a new GAVDependencyTree
+     * and map the GAV to the GAVDependencyTree
+     *
+     * @param map
+     * @param gav
+     * @return
+     */
+    private GAVDependencyTree addGAVDependencyTreeToMapper(Map<GAV, GAVDependencyTree> map, GAV gav) {
+        return map.computeIfAbsent(gav, (k) -> new GAVDependencyTree(k));
+    }
+
+    private GAV generateGAV(ProjectVersionRef ref) {
+        return new GAV(ref.getGroupId(), ref.getArtifactId(), ref.getVersionString());
+    }
+
+    private DiscoveryResult getDiscoveryResult(File tempDir, File repoDir, ProjectVersionRef ref)
+            throws GalleyMavenException, URISyntaxException, CartoDataException {
+
+        MavenPomReader mavenPomReader = carto.getGalley().getPomReader();
+        MavenPomView pomView = mavenPomReader.read(ref, getRepoLocations(tempDir, repoDir));
+
+        Location location = new SimpleLocation("file:" + tempDir.getAbsolutePath());
+        URI src = new URI(location.getUri());
+
+        DiscoveryConfig disConf = new DiscoveryConfig(src);
+        disConf.setIncludeBuildSection(false);
+        disConf.setIncludeManagedDependencies(false);
+        disConf.setIncludeManagedPlugins(false);
+
+        MavenModelProcessor processor = new MavenModelProcessor();
+        return processor.readRelationships(pomView, src, disConf);
+    }
+
+    private List<Location> getRepoLocations(File cacheDir, File repoDir) {
+
+        Location location = new SimpleLocation("file:" + cacheDir.getAbsolutePath());
+
+        List<Location> repoLocations = new LinkedList<>();
+        // add local location
+        repoLocations.add(location);
```

Comment:
Yup can do! You know, when you're working on the code for weeks, I get a little (ok a lot) bit lazy. :sweat_smile:  Thanks for the suggestion! Will change it!